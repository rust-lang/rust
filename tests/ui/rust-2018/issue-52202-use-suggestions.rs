//@ edition:2018

// The local `use` suggestion should start with `crate::` (but the
// standard-library suggestions should not, obviously).

mod plumbing {
    pub struct Drain;
}

fn main() {
    let _d = Drain {};
    //~^ ERROR cannot find struct, variant or union type `Drain` in this scope
}
