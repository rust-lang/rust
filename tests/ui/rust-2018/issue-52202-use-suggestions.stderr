error[E0422]: cannot find struct, variant or union type `Drain` in this scope
  --> $DIR/issue-52202-use-suggestions.rs:11:14
   |
LL |     let _d = Drain {};
   |              ^^^^^ not found in this scope
   |
help: consider importing one of these structs
   |
LL + use crate::plumbing::Drain;
   |
LL + use std::collections::binary_heap::Drain;
   |
LL + use std::collections::hash_map::Drain;
   |
LL + use std::collections::hash_set::Drain;
   |
     and 3 other candidates

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0422`.
